### DISCUSSION TITLE
Load all `.rego` files from a directory and evaluate it via golang rego library

### DISCUSSION THREAD

**Discussion by Abhishek-kumar09** [open-policy-agent/community#419] [OPA and Rego] [answered]
Hello, 
Actually, I am helping my team integrate OPA for our policy evaluation, some [initial draft work is here.](https://github.com/meshery/meshery/pull/7654)

If you see, I am successfully able to load a policy file and evaluate it via rego engine, but the complete usecase looks something like this, where there is a `"policies"` directory where all `.rego` files will be placed and against the input we would send some `rego.Query("...")` that will run the rego evaluation engine.

Though there are bunch of resources on OPA CLI, I'm finding trouble implementing it in golang.
How can I read all the rego files from a folder and send it to evaluation engine for processing?

Thanks.

**Comment by anderseknert** [comment, marked as answer]
Hi there! This what you're looking for, I think:

https://pkg.go.dev/github.com/open-policy-agent/opa/rego#Load

**Comment by jpreese** [comment]
You may also be able to get some inspiration from Conftest, which works in a similar way: https://github.com/open-policy-agent/conftest/blob/master/policy/engine.go#L65

**Comment by Abhishek-kumar09** [comment]
Thanks @anderseknert. I'm trying the solution and looking at the reference 😃

**Comment by Abhishek-kumar09** [comment]
Thanks that works, if someone comes by this: here is the initial setup: https://github.com/meshery/meshkit/pull/311
